assignments/lambda-classes.js

```diff
@@ -1 +1,115 @@
-// CODE here for your Lambda Classes
+//base class
+
+class Person {
+  constructor(props) {
+    this.name = props.name;
+    this.age = props.age;
+    this.location = props.location;
+    this.gender = props.gender;
+    this.catchPhrase = props.catchPhrase;
+  }
+  speak(){
+    return `Hello my name is ${this.name}, I am from ${this.location}`
+  }
+}
+
+class Instructor extends Person {
+  constructor(instructorProps) {
+    super(instructorProps);
+    this.specialty = instructorProps.specialty;
+    this.favLanguage = instructorProps.favLanguage;
+    }
+    demo(subject){
+    return  `Today we are learning about ${subject}`
+    }
+    grade(student, subject){
+      return `${student.name} receives a perfect score on ${subject}`
+    }
+}
+
+class Student extends Person {
+  constructor(studentParams) {
+    super(studentParams);
+    this.previousBackground = studentParams.previousBackground;
+    this.className = studentParams.className;
+    this.favSubjects = studentParams.favSubjects;
+  }
+
+  listsSubjects(skill1, skill2, skill3){
+    return `${skill1}, \n${skill2}, \n${skill3} `;
+  }
+  PRAssignment (subject) {
+    return `${this.name} has submitted a PR for ${subject}`
+  }
+  sprintChallenge(subject){
+    return `${this.name} has begun sprint challenge on ${subject}`
+  }
+
+}
+
+class ProjectManager extends Instructor {
+  constructor(pmParams) {
+    super(pmParams);
+    this.gradClassName = pmParams.gradClassName;
+    this.favInstructor = pmParams.favInstructor;
+  }
+
+  standUp () {
+  return  `${this.name} announces to ${this.gradClassName}, @channel standy times!`
+  }
+  debugsCode(student){
```

Comment:
This method should take in two parameters, student and subject, as the last template string literal should be `${subject}`.

But I also like the creative solution you came up with.